packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js

```diff
@@ -12,143 +11,42 @@ import Carousel from '../ProductImageCarousel';
-    return { key, value };
-};
-
 const ProductFullDetail = props => {
-    const [{ isAddingItem }, { addItemToCart }] = useCartContext();
-
-    // Props.
     const { product } = props;
 
-    // State.
-    const [quantity, setQuantity] = useState(INITIAL_QUANTITY);
-    const [optionSelections, setOptionSelections] = useState(
-        INITIAL_OPTION_SELECTIONS
-    );
-    const derivedOptionCodes = deriveOptionCodesFromProduct(product);
-    const [optionCodes] = useState(derivedOptionCodes);
-
-    // Members.
-    const { amount: productPrice } = product.price.regularPrice;
-    const classes = mergeClasses(defaultClasses, props.classes);
-    const isMissingOptions = getIsMissingOptions(product, optionSelections);
-    const mediaGalleryEntries = getMediaGalleryEntries(
-        product,
-        optionCodes,
-        optionSelections
-    );
+    const talonProps = useProductFullDetail({
+        product
+    });
 
-    // Event handlers.
-    const handleAddToCart = useCallback(() => {
-        const payload = {
-            item: product,
-            productType: product.__typename,
-            quantity
-        };
+    const {
+        handleAddToCart,
+        handleSelectionChange,
+        handleSetQuantity,
+        isAddToCartDisabled,
+        mediaGalleryEntries,
+        productDetails,
+        quantity
+    } = talonProps;
 
-        if (isProductConfigurable(product)) {
-            appendOptionsToPayload(payload, optionSelections, optionCodes);
-        }
-
-        addItemToCart(payload);
-    }, [addItemToCart, optionCodes, optionSelections, product, quantity]);
-
-    const handleSelectionChange = useCallback(
-        (optionId, selection) => {
-            // We must create a new Map here so that React knows that the value
-            // of optionSelections has changed.
-            const newOptionSelections = new Map([...optionSelections]);
-            newOptionSelections.set(optionId, Array.from(selection).pop());
-            setOptionSelections(newOptionSelections);
-        },
-        [optionSelections]
-    );
+    const classes = mergeClasses(defaultClasses, props.classes);
 
     return (
         <Form className={classes.root}>
             <section className={classes.title}>
-                <h1 className={classes.productName}>{product.name}</h1>
+                <h1 className={classes.productName}>{productDetails.name}</h1>
                 <p className={classes.productPrice}>
                     <Price
-                        currencyCode={productPrice.currency}
-                        value={productPrice.value}
+                        currencyCode={productDetails.price.currency}
+                        value={productDetails.price.value}
                     />
                 </p>
             </section>
             <section className={classes.imageCarousel}>
-                <Carousel
-                    images={mediaGalleryEntries.value}
-                    key={mediaGalleryEntries.key}
```

Comment:
`key` didn't seem used so I deleted the code generating it and just returned the computed entries.

Reply:
Just from [looking quickly](https://github.com/magento/pwa-studio/pull/680), I think it's safe to say there was a valid reason to calculate that key before. Did something change so that it's no longer necessary?